src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -968,8 +975,18 @@ namespace xclcpuemhal2 {
     if (mLogStream.is_open()) {
       mLogStream << __func__ << ", " << std::this_thread::get_id() << std::endl;
     }
-    if(!sock)
+    if (!sock) {
+      PRINTENDFUNC
+      if (mIsKdsSwEmu && mSWSch && mCore)
+      {
+        mSWSch->fini_scheduler_thread();
+        delete mCore;
+        mCore = NULL;
```

Comment:
in C++ NULL should be replaced by nullptr

Reply:
changed to nullptr